src/components/MultiStepIndicator/components/StepIndicator/StepIndicator.tsx

```diff
@@ -18,48 +17,79 @@ import Clickable from "../../../../components/Clickable/Clickable";
 import { SIZES, MULTI_STEP_TYPES, STEP_STATUSES } from "../../MultiStepConstants";
 import styles from "./StepIndicator.module.scss";
 import classNames from "classnames";
+import { SubIcon, VibeComponentProps } from "../../../../types";
+import { IconType } from "../../../Icon/IconConstants";
 
 const KEYS = [keyCodes.ENTER, keyCodes.SPACE];
 
-const StepCircleDisplay = ({
+interface StepCircleDisplayProps {
+  status: STEP_STATUSES;
+  isFulfilledStepDisplayNumber: boolean;
+  fulfilledStepIcon: SubIcon;
+  fulfilledStepIconType: IconType.SVG | IconType.ICON_FONT;
+  stepNumber: number;
+  size: SIZES;
+}
+
+const StepCircleDisplay: React.FC<StepCircleDisplayProps> = ({
   status,
   isFulfilledStepDisplayNumber,
   fulfilledStepIcon,
   fulfilledStepIconType,
-  stepNumber,
-  size
+  stepNumber
 }) => {
-  return status === STEP_STATUSES.FULFILLED && !isFulfilledStepDisplayNumber ? (
-    <Icon
-      icon={fulfilledStepIcon}
-      className={classNames(styles.numberContainerTextCheckIcon, { [styles.compact]: size === SIZES.COMPACT })}
-      iconLabel={STEP_STATUSES.FULFILLED}
-      iconType={fulfilledStepIconType}
-      ignoreFocusStyle
-      clickable={false}
-      ariaHidden={true}
-    />
-  ) : (
-    stepNumber
+  return (
+    <>
+      {status === STEP_STATUSES.FULFILLED && !isFulfilledStepDisplayNumber ? (
+        <Icon
+          icon={fulfilledStepIcon}
+          className={classNames(styles.numberContainerTextCheckIcon)}
+          iconLabel={STEP_STATUSES.FULFILLED}
+          iconType={fulfilledStepIconType}
+          ignoreFocusStyle
+          clickable={false}
+          ariaHidden={true}
+        />
+      ) : (
+        stepNumber
+      )}
+    </>
   );
 };
 
-const StepIndicator = ({
+export interface StepIndicatorProps extends VibeComponentProps {
+  status: STEP_STATUSES;
+  titleText: string;
+  subtitleText?: string;
+  stepNumber: number;
+  stepComponentClassName?: string;
+  type?: MULTI_STEP_TYPES;
+  fulfilledStepIcon?: SubIcon;
+  fulfilledStepIconType?: IconType.SVG | IconType.ICON_FONT;
+  isFulfilledStepDisplayNumber?: boolean;
+  onClick?: (stepNumber: number) => void;
+  isFollowedByDivider?: boolean;
+  stepDividerClassName?: string;
+  isVertical?: boolean;
+  size?: SIZES;
+}
+
+const StepIndicator: React.FC<StepIndicatorProps> = ({
   stepComponentClassName,
-  stepNumber,
-  status,
-  titleText,
-  subtitleText,
-  type,
-  fulfilledStepIcon,
-  fulfilledStepIconType,
-  isFulfilledStepDisplayNumber,
-  onClick,
-  isFollowedByDivider,
+  stepNumber = 1,
```

Comment:
same comment about double check that the default values of the props did not changed by mistake :)

Reply:
Yep checked everything again, seems correct :)
although the prop is set to be required, I wanted to keep the existing behavior (it is currently set as required but has default)
```
StepIndicator.propTypes = {
  ...
  stepNumber: PropTypes.number.isRequired,
  ...
}

StepIndicator.defaultProps = {
  ...
  stepNumber: 1,
  ...
}
```

Reply:
remove required